plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt

```diff
@@ -114,14 +114,27 @@ abstract class MavenPublishBaseExtension(
 
     project.plugins.apply(SigningPlugin::class.java)
     project.gradleSigning.setRequired(Callable { !project.versionIsSnapshot })
-    project.gradleSigning.sign(project.gradlePublishing.publications)
 
     val inMemoryKey = project.findOptionalProperty("signingInMemoryKey")
     if (inMemoryKey != null) {
       val inMemoryKeyId = project.findOptionalProperty("signingInMemoryKeyId")
       val inMemoryKeyPassword = project.findOptionalProperty("signingInMemoryKeyPassword") ?: ""
       project.gradleSigning.useInMemoryPgpKeys(inMemoryKeyId, inMemoryKey, inMemoryKeyPassword)
     }
+
+    // TODO: replace with the following line after https://github.com/gradle/gradle/issues/21857 is fixed
+    //  project.gradleSigning.sign(project.gradlePublishing.publications)
+    project.gradlePublishing.publications.withType(MavenPublication::class.java).all { publication ->
+      val task = project.tasks.findByName("sign${publication.name.capitalize()}Publication")
+      if (task == null) {
+        project.gradleSigning.sign(publication)
+      }
+    }
```

Comment:
Only call sign on a publication if there is no signing task for it yet which fixes the issue when other plugins already called it.